Introduced backend v2

purbanski-deftcode opened the pull request.

Commit:
Introduced backend v2

grzegorz-bielski commented on vue.config.js:
```diff
@@ -1,10 +1,11 @@
-                secure: false
-            }
-        }
+  devServer: {
```
Maybe we can remove it... I was using it when docker-compose was still a WIP.

grzegorz-bielski commented on src/modules/CanHelp/CanHelp.ts:
```diff
@@ -54,10 +54,10 @@ export namespace CanHelp {
           )
         ),
-        O.map(([helper, requests]) => ({ helper, requests })),
+        O.map(([helper, request]) => ({ helper, request })),
```
I'm afraid we cannot merge it to master like that, as this will break the compatibility with existing backend.

grzegorz-bielski commented on src/modules/CanHelp/Outlet.vue:
```diff
@@ -155,7 +155,7 @@ export default class CanHelpOutlet extends Vue {
   towns = ["Wszystkie"];
   selectedTown = "";
-  path = "/api/requests/";
+  path = "http://localhost:5011/help-request";
```
Again, we cannot merge it, as this breaks current compatibility, and we shouldn't really use hardcoded ports.
I think the problem lies within our routing. Is nest's API served under `/api` ?

lukaszmindpal replied:
yes let's try /api/

grzegorz-bielski commented on .gitmodules:
```diff
@@ -1,3 +1,3 @@
-	url = https://github.com/marcinlovescode/uszyj-maseczke.git
+[submodule "pomocnicy-online-backend"]
+	path = pomocnicy-online-backend
+	url = https://github.com/purbanski-deftcode/pomocnicy-online-backend.git
```
For now let's add another submodule instead of changing the current one

grzegorz-bielski commented on docker-compose.yml:
```diff
@@ -15,17 +15,17 @@ services:
+
   web:
-    build: ./backend
+    build: ./pomocnicy-online-backend
```
Let's use another docker-compose file for the transition period, which overrides `web`

lukaszmindpal replied:
We can add another service like `backend`, you can copy-paste web section and change the name/parameters

grzegorz-bielski reviewed (changes requested):
Thank you for the hard work on backendPrim, we really appreciate it!

Unfortunately, this is a breaking change, which makes the app not usable with the current backend system, which is still operational and we haven't migrated the data. 
 
Let's do it with the spirit of progressive enhancement!
Ideally, there would be a second make script that fires the project with a new backend and there are no breaking changes. At least for now.

Commit:
Removed submodules

Commit:
Cleaned up submodules

Commit:
Revert "Introduced backend v2"

Commit:
Removed breaking changes

purbanski-deftcode commented:
@grzegorz-bielski please re-review. What has changed:
- removed the breaking changes, now it is compatible with old backend
- adjusted nginx to route `/api/v2/` to new backend
- adjusted docker-compose to serve both old and new backend

Notes:
- tables in DB are not compatible, only api endpoints are, but as discussed with @sojek we will move the data from old tables to new ones after transition

Open question:
- Not sure how to approach the transition itself, for now the idea is to change api url from `/api/` to `/api/v2/` in file with the `API_URL` constant whenever new backend is on production, you have any idea how to do this in smoother way?

grzegorz-bielski commented:
>     * Not sure how to approach the transition itself, for now the idea is to change api url from `/api/` to `/api/v2/` in file with the `API_URL` 

Sounds fine! Sorry for the late response 😓
